application/src/main/java/org/opentripplanner/updater/alert/GtfsRealtimeAlertsUpdater.java

```diff
@@ -63,32 +64,28 @@ public String toString() {
   }
 
   @Override
-  protected void runPolling() {
-    try {
-      final FeedMessage feed = otpHttpClient.getAndMap(
-        URI.create(url),
-        this.headers.asMap(),
-        FeedMessage.PARSER::parseFrom
-      );
+  protected void runPolling() throws InterruptedException, ExecutionException {
+    final FeedMessage feed = otpHttpClient.getAndMap(
+      URI.create(url),
+      this.headers.asMap(),
+      FeedMessage.PARSER::parseFrom
+    );
 
-      long feedTimestamp = feed.getHeader().getTimestamp();
-      if (feedTimestamp == lastTimestamp) {
-        LOG.debug("Ignoring feed with a timestamp that has not been updated from {}", url);
-        return;
-      }
-      if (feedTimestamp < lastTimestamp) {
-        LOG.info("Ignoring feed with older than previous timestamp from {}", url);
-        return;
-      }
+    long feedTimestamp = feed.getHeader().getTimestamp();
+    if (feedTimestamp == lastTimestamp) {
+      LOG.debug("Ignoring feed with a timestamp that has not been updated from {}", url);
+      return;
+    }
+    if (feedTimestamp < lastTimestamp) {
+      LOG.info("Ignoring feed with older than previous timestamp from {}", url);
+      return;
+    }
 
-      // Handle update in graph writer runnable
-      saveResultOnGraph.execute(context ->
-        updateHandler.update(feed, context.gtfsRealtimeFuzzyTripMatcher())
-      );
+    // Handle update in graph writer runnable
+    saveResultOnGraph
+      .execute(context -> updateHandler.update(feed, context.gtfsRealtimeFuzzyTripMatcher()))
+      .get();
```

Comment:
This is a good change, but we need to be able to turn this off in case it have some side-effects we have missed. This is critical code, and it is possible that we have not understood all implications. So, it would be nice to introduce a `OTPFeature` to be able to turn this the synchronisation _off_. This need to be done for all calls to `Feature#get()` introduced in this PR. 

```suggestion
    var f = saveResultOnGraph.execute(context -> updateHandler.update(feed, context.gtfsRealtimeFuzzyTripMatcher()));
      if(OTPFeature.WaitOnPullingUpdatersBeforeResheduling.isOn()) {
        f.get();
      }
```

Reply:
I will add a method to the updater and a unit test to do that, such that the behaviour is testable.

Reply:
> This is a good change, but we need to be able to turn this off in case it have some side-effects we have missed.

In the alternative I suggested, this could be accomplished by an additional condition

```
if (OTPFeature.AwaitPollingUpdaterCompletion.isOff() || status == null || status.isDone()) {
        status = runPolling();
}
```

Though I think safety measures (and maybe even the OTPOption) are less necessary in an implementation where there are no blocking calls.

Reply:
(@t2gran sorry, I accidentally edited your comment instead of hitting "quote reply". I think I restored your comment to its original form.)